apps/src/templates/SortByNameDropdown.jsx

```diff
@@ -1,38 +1,44 @@
-import {useDispatch} from 'react-redux';
+import {connect} from 'react-redux';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, {useCallback} from 'react';
 import i18n from '@cdo/locale';
 import {setSortByFamilyName} from '@cdo/apps/templates/currentUserRedux';
+import UserPreferences from '@cdo/apps/lib/util/UserPreferences';
+
+const FAMILY_NAME = 'familyName';
+const DISPLAY_NAME = 'displayName';
 
 function SortByNameDropdown({
   sortByStyles,
   selectStyles,
   sectionId,
   unitName,
   source,
+  isSortedByFamilyName,
+  setSortByFamilyName,
 }) {
-  const dispatch = useDispatch();
+  const onSortByFamilyNameChange = useCallback(
+    e => {
+      const newSort = e.target.value === FAMILY_NAME;
+      new UserPreferences().setSortByFamilyName(newSort);
+      setSortByFamilyName(newSort, sectionId, unitName, source);
+    },
+    [sectionId, unitName, source, setSortByFamilyName]
+  );
+
+  const selectedValue = isSortedByFamilyName ? FAMILY_NAME : DISPLAY_NAME;
   return (
     <div>
       <div style={sortByStyles}>{i18n.sortBy()}</div>
       <select
         name="familyNameSort"
         aria-label={i18n.sortBy()}
         style={selectStyles}
-        defaultValue={i18n.displayName}
-        onChange={e => {
-          dispatch(
-            setSortByFamilyName(
-              e.target.value === 'true',
-              sectionId,
-              unitName,
-              source
-            )
-          );
-        }}
+        value={selectedValue}
+        onChange={onSortByFamilyNameChange}
       >
-        <option value={false}>{i18n.displayName()}</option>
```

Comment:
This was doing some weird conversions between bool and string. I think changing it to descriptive strings makes this much clearer.